### PR TITLE
Fix highlight text color in combobox and toolbutton.

### PR DESCRIPTION
**Pull request by xuzhao9**
This PR fixes incorrect text color in ToolButton and ComboBox widgets.

For ToolButton, this PR updates the text color when the widget is pressed down (`sunken`), making the text easier to recognize.

For ComboBox, this PR also updates the text color when the widget has focus and is `sunken`.

Note that we only need to change the text color when the style is not `flat`.

### CODE REVIEW

File: kstyle/lightlystyle.cpp [hunk 1]
```diff
@@ -4893,7 +4892,7 @@ namespace Lightly
             if( hasFocus && sunken ) textRole = QPalette::HighlightedText;
             else textRole = QPalette::WindowText;
 
-        } else if( hasFocus ) textRole = QPalette::HighlightedText;
+        } else if( option->state & State_HasFocus || sunken) textRole = QPalette::HighlightedText;
```
Comment (xuzhao9) [inline kstyle/lightlystyle.cpp:4895]: Can't use `hasFocus` here because we want to change the text color when state is `State_HasFocus` AND mouseover.

**Comment by Luwx** [comment]
Thanks for the fix! The changes look good. Can you resolve the conflict so that this can be merged?

**Comment by xuzhao9** [comment]
> Thanks for the fix! The changes look good. Can you resolve the conflict so that this can be merged?

Thanks for the response! I have fixed the merge conflict.

**Comment by xuzhao9** [comment]
@Luwx  Please let me know what else I need to do to get this PR merged. Thanks!

**Comment by Luwx** [comment]
Great! Merging

**Pull request merged**